http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala

```diff
@@ -193,6 +207,32 @@ class CollectorService[F[_]: Sync](
         )
     }
 
+  /** Builds the appropriate http response when dealing with click redirects. */
+  def buildRedirectHttpResponse(queryParams: Map[String, String], headers: Headers): Response[F] = {
+    val targetUri = for {
+      target <- queryParams.get("u")
+      uri    <- Uri.fromString(target).toOption
+      _ = uri if redirectTargetAllowed(uri)
```

Comment:
It filters out the uri if `redirectTargetAllowed` return false.

Reply:
I don't get how :thinking: 
`<-` is used for the `.flatMap`, e.g. `uri <- if(redirectTargetAllowed(uri)) Some(uri) else None` but how does the `=` short-circuits in case of `false` ?

Reply:
I've realized `_ = uri` part isn't necessary. It can be written like this:
```scala
    val targetUri = for {
      target <- queryParams.get("u")
      uri    <- Uri.fromString(target).toOption
      if redirectTargetAllowed(uri)
    } yield uri
```
`if` can be used in comprehension. Similar to `<-` is converted to `flatMap`, `if` is converted to `withFilter`. There is a good [SO answer](https://stackoverflow.com/a/7949907) explains this.